build_test/lib/src/record_logs.dart

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:build/src/builder/logging.dart';
+import 'package:matcher/matcher.dart';
+import 'package:logging/logging.dart';
+
+/// Executes [run] with a new [Logger], returning the resulting log records.
+///
+/// ```dart
+/// test('should log "uh oh!"', () async {
+///   final logs = recordLogs(() => runBuilder());
+///   expect(logs, emitsInOrder([
+///     anyLogOf('uh oh!'),
+///   ]);
+/// });
+/// ```
+Stream<LogRecord> recordLogs(dynamic run(), {String name: ''}) {
+  final logger = new Logger(name);
+  Timer.run(() async {
+    await scopeLog(run, logger);
+    logger.clearListeners();
+  });
+  return logger.onRecord;
+}
+
+/// Matches [LogRecord] of any level whose message is [messageOrMatcher].
+///
+/// ```dart
+/// anyLogOf('Hello World)';     // Exactly match 'Hello World'.
+/// anyLogOf(contains('ERROR')); // Contains the sub-string 'ERROR'.
+/// ```
+Matcher anyLogOf(dynamic messageOrMatcher) =>
```

Comment:
I could imagine `greaterThan` and `lessThan` optional args here (would be `Level`s), but probably not really worth the complexity.

Reply:
Yeah. I'll wait until needed ;)